### PR TITLE
fix(segmentation): convert OpenCV-decoded image from BGR to RGB before model transform

### PR DESCRIPTION
**Pull request by AndreaMariani-AM**
## Summary

This PR adds an explicit channel-order conversion in tissue segmentation:

- `cv2.imdecode(...)` output is converted from **BGR** to **RGB** before tensor conversion and model transforms.

### Change

In `src/lazyslide/segmentation/_tissue.py`, after JPEG decode:

```python
img = cv2.imdecode(img, 1)
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
img = torch.tensor(img).permute(2, 0, 1)
```

## Why

OpenCV standard image channel order is **BGR**.  
The segmentation model transforms in `lazyslide-models` expect **RGB** semantics:

- `PathProfiler` CLAHE uses `cv2.COLOR_RGB2HSV` / `cv2.COLOR_HSV2RGB`.
- `HEST` and `GrandQC` transforms use RGB-ordered normalization values.

Without explicit conversion, there is a risk of channel mismatch (blue/red swap), which can affect preprocessing and segmentation quality.

## Notes for maintainers

Please verify expected channel convention end-to-end in the pipeline (WSI reader output -> JPEG simulation -> transforms), but this change aligns the OpenCV decode output with model transform expectations.

## Scope

- No logic refactor.
- No model/interface changes.
- Only channel-order correction before transform.

### CODE REVIEW

**Comment by Mr-Milk** [comment]
@AndreaMariani-AM Thanks for the fix!
